### DISCUSSION TITLE
Change a default (fallback) setting to prevent issues like mine?

### DISCUSSION THREAD

**Discussion by erdajt** [phpactor/phpactor#2684] [Polls]
Today, I encountered an issue while setting up **Phpactor** in ***Neovim***. After installing it through Mason and configuring it in `lspconfig`, it failed to mount, and I couldn't find any errors or warnings despite several debugging attempts and extensive research on various forums.

The breakthrough came when I noticed a message on `:LspInfo` stating `root directory: Not found`. This indicated that **Phpactor** could not locate the root directory of the project, usually identified by the presence of a composer.json file.

To resolve this, I modified the setup as follows:

```lua
require('lspconfig').phpactor.setup({
  root_dir = function(fname)
    return require('lspconfig/util').find_git_ancestor(fname) or vim.fn.getcwd()
  end
})
```

This configuration sets the root directory to the nearest Git ancestor, if available, or defaults to the current working directory. While initially testing the LSP on standalone PHP files without a project structure led to this oversight,

Finally to make my point:
### I believe this root directory determination should be the default setting to prevent similar issues.

**Comment by dantleech** [comment]
What's different about your project? You have no `composer.json` file? Note that Phpactor depends on this file for some functionality. But otherwise, how this is determined is outside of the scope of this project and lies more with `lspconfig` - how do other PHP langauge servers determine their root path in that project?

**Reply by erdajt** [reply]
All I'm suggesting is:

*if phpactor cannot find `composer.json` default to a fallback file.*

I understand that ***some*** functionality of phpactor is reliant on the contents of `composer.json` but, an *LSP* to loose all functionality if a specific dependency/package manager (composer) is not used isn't exactly practical.

**Reply by dantleech** [reply]
Which is fine, but that' not Phpactor's job, n it's the jobnof the LSP client, anr in this case lspconfig
